### PR TITLE
[Infra] Bump python version to 3.10

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
Bump python version to 3.10 to better align witn our TPU VM minimum version.

Test Plan:
N/A

### CODE REVIEW

File: scripts/build_torch_wheels.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -x  # Display commands being run.
 PYTHON_VERSION=$1
 RELEASE_VERSION=$2  # rX.Y or nightly
 BUILD_CPP_TESTS="${3:-0}"
-DEFAULT_PYTHON_VERSION=3.8
+DEFAULT_PYTHON_VERSION=3.10
```
Comment (will-cromar) [inline scripts/build_torch_wheels.sh:9]: I don't think this script is used anymore

File: infra/tpu-pytorch-releases/dev_images.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ variable "dev_images" {
   type = list(object({
     accelerator    = string
     arch           = optional(string, "amd64")
-    python_version = optional(string, "3.8")
+    python_version = optional(string, "3.10")
```
Comment (will-cromar) [inline infra/tpu-pytorch-releases/dev_images.tf:5]: Do you mind just adding `python_version = "3.10"` to the images you want to update (e.g. [here](https://github.com/pytorch/xla/blob/46e2230012c37ef0cdaf4b9fa7bde1d61e2ebe06/infra/tpu-pytorch-releases/dev_images.auto.tfvars#L9)) instead of updating the default? IMO we should remove this default and explicitly mark the python version we want for each artifact, but that's a separate PR

File: infra/ansible/development.Dockerfile [hunk 3]
```diff
@@ -17,3 +17,8 @@ ENV TAGS="bazel,configure_env,install_deps"
 ARG ansible_vars
 RUN ansible-playbook playbook.yaml -e "stage=build" -e "${ansible_vars}" --tags "${TAGS}"
 RUN ansible-playbook playbook.yaml -e "stage=release" -e "${ansible_vars}" --tags "${TAGS}"
+
+RUN useradd --create-home --shell /bin/bash ptxla
```
Comment (will-cromar) [inline infra/ansible/development.Dockerfile:21]: Can you move this to a separate PR? This is much riskier than updating the Python version.

Comment (will-cromar) [reply infra/ansible/development.Dockerfile:21]: I'll see if there's a way to use the TPU as a non-root user in a container in the meantime.

**Review by will-cromar — COMMENTED** [review]
To test the terraform changes, you can run `terraform plan` from the `tpu-pytorch-releases` directory and post the output here.

File: .devcontainer/tpu-contributor/devcontainer.json [hunk 4]
```diff
@@ -22,5 +22,6 @@
         "ms-python.python"
       ]
     }
-  }
+  },
+  "containerUser": "ptxla"
```
Comment (yeounoh) [inline .devcontainer/tpu-contributor/devcontainer.json:26]: This may break the builds or test executions, etc.. I think we may need sudo/root privilege for some of the operations. cc @will-cromar

File: infra/ansible/development.Dockerfile (hunk 3 above)
Comment (yeounoh) [reply infra/ansible/development.Dockerfile:21]: ah, +1 https://github.com/pytorch/xla/pull/6685/files#r1516655463

File: .devcontainer/tpu-contributor/devcontainer.json (hunk 4 above)
Comment (alanwaketan) [reply .devcontainer/tpu-contributor/devcontainer.json:26]: Will remove that.

File: infra/ansible/development.Dockerfile (hunk 3 above)
Comment (alanwaketan) [reply infra/ansible/development.Dockerfile:21]: Will remove that.

File: infra/tpu-pytorch-releases/dev_images.tf (hunk 2 above)
Comment (alanwaketan) [reply infra/tpu-pytorch-releases/dev_images.tf:5]: which means I need to discard all the changes in this PR?

**Comment by alanwaketan** [comment]
@will-cromar @yeounoh I discarded all changes and only change the file Will suggested?

**Review by will-cromar — APPROVED** [review]
Thanks! Should the CUDA image also be Python 3.10?

**Comment by alanwaketan** [comment]
> Thanks! Should the CUDA image also be Python 3.10?

I have no ideas what's the default python version for gpu vm. cc @vanbasten23 do you have any recommendations?
